express/blocks/gen-ai-cards/gen-ai-cards.js

```diff
@@ -0,0 +1,219 @@
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      handleGenAISubmit(genAIForm, ctaLinks[0].href);
+    }
+  });
+
+  genAIForm.addEventListener('submit', (e) => {
+    e.preventDefault();
+    handleGenAISubmit(genAIForm, ctaLinks[0].href);
+  });
+
+  return genAIForm;
+}
+
+function removeLazyAfterNeighborLoaded(image, lastImage) {
+  if (!image || !lastImage) return;
+  lastImage.onload = (e) => {
+    if (e.eventPhase >= Event.AT_TARGET) {
+      image.querySelector('img').removeAttribute('loading');
+    }
+  };
+}
+
+async function decorateCards(block, { actions }) {
+  const cards = createTag('div', { class: 'gen-ai-cards-cards' });
+  let searchBranchLinks;
+
+  await import(`${getLibs()}/features/placeholders.js`).then(async (mod) => {
+    searchBranchLinks = await mod.replaceKey('search-branch-links', getConfig());
+    return mod.replaceKey();
+  });
+
+  actions.forEach((cta, i) => {
+    const {
+      image,
+      ctaLinks,
+      text,
+      title,
+    } = cta;
+    const card = createTag('div', { class: 'card' });
+    const linksWrapper = createTag('div', { class: 'links-wrapper' });
+    const mediaWrapper = createTag('div', { class: 'media-wrapper' });
+    const textWrapper = createTag('div', { class: 'text-wrapper' });
+
+    card.append(textWrapper, mediaWrapper, linksWrapper);
+    if (image) {
+      mediaWrapper.append(image);
+      if (i > 0) {
+        const lastImage = actions[i - 1].image?.querySelector('img');
+        removeLazyAfterNeighborLoaded(image, lastImage);
+      }
+    }
+
+    const hasGenAIForm = promptTokenRegex.test(ctaLinks?.[0]?.href);
+
+    if (ctaLinks.length > 0) {
+      if (hasGenAIForm) {
+        const genAIForm = buildGenAIForm(cta);
+        card.classList.add('gen-ai-action');
+        card.append(genAIForm);
+        linksWrapper.remove();
+      } else {
+        const a = ctaLinks[0];
+        const btnUrl = new URL(a.href);
+        if (searchBranchLinks?.replace(/\s/g, '').split(',').includes(`${btnUrl.origin}${btnUrl.pathname}`)) {
+          btnUrl.searchParams.set('q', cta.text);
+          btnUrl.searchParams.set('category', 'templates');
+          a.href = decodeURIComponent(btnUrl.toString());
+        }
+        a.removeAttribute('title');
+        linksWrapper.append(a);
+      }
+    }
+
+    const titleText = decorateTextWithTag(title, { tagT: 'sup', baseClass: 'cta-card-title' });
+    textWrapper.append(titleText);
+    const desc = createTag('p', { class: 'cta-card-desc' });
+    desc.textContent = text;
+    textWrapper.append(desc);
+
+    cards.append(card);
+  });
+
+  block.append(cards);
+}
+
+function constructPayload(block) {
+  const rows = Array.from(block.children);
+  block.innerHTML = '';
+  const headingDiv = rows.shift();
+
+  const payload = {
+    heading: headingDiv.querySelector('h2, h3, h4, h5, h6')?.textContent?.trim(),
+    subHeadings: headingDiv.querySelectorAll('p:not(.button-container)'),
```

Comment:
can you also add checks for milo buttons? - `p:not(.button-container), a:not(.con-button)`